app/src/main/kotlin/com/github/gotify/CoilHandler.kt

```diff
@@ -0,0 +1,62 @@
+package com.github.gotify
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.drawable.BitmapDrawable
+import coil.ImageLoader
+import coil.annotation.ExperimentalCoilApi
+import coil.disk.DiskCache
+import coil.request.ImageRequest
+import com.github.gotify.api.CertUtils
+import com.github.gotify.client.model.Application
+import okhttp3.OkHttpClient
+import org.tinylog.kotlin.Logger
+import java.io.IOException
+
+internal class CoilHandler(private val context: Context, private val settings: Settings) {
+    private val imageLoader = makeImageLoader()
```

Comment:
The coil docs say that the loader should be a singleton. Maybe we can use the provided singleton from coil? https://coil-kt.github.io/coil/image_loaders/#singleton-vs-dependency-injection

Reply:
I've oriented towards this https://coil-kt.github.io/coil/image_loaders/#image-loaders.

> Coil performs best when you create a single `ImageLoader` and share it throughout your app. This is because each `ImageLoader` has its own memory cache, disk cache, and `OkHttpClient`.

Reply:
Oke, but we currently have two image loaders. One is instantiated in the WebSocketService and one in the MessageActivity. Maybe there could be race conditions when they use the same disk cache.

Reply:
Ahh the settings dependency again .___. I've tried a little but I'm not really satisfied with this solution.
```kotlin
private var holder: Pair<SSLSettings, ImageLoader>? = null;

@Synchronized
fun get(context: Context): ImageLoader {
    val newSettings = Settings(context).sslSettings()
    val copy = holder;
    if (copy != null && copy.first == settings) {
        return copy.second
    }
    val loader = makeImageLoader(context, newSettings)
    holder = (newSettings to loader)
    return loader
}
```
basically every caller passes the context and we check if the cached image loader is ok.

What do you think? I'd be okay with ignoring this and have two instances, as it's probably not much of a problem.

Reply:
Yes.